inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -450,16 +450,13 @@ bool layout_optimizer::convolution_b_fs_zyx_fsv16_opt(layout const &input_layout
                                                       const layout &weights_layout,
                                                       std::shared_ptr<const convolution> conv) {
     // A set of rules that define when b_fs_zyx_fsv16 mem format can be used
-    if ((input_layout.format == format::bfzyx ||
-         input_layout.format == format::b_fs_zyx_fsv16 ||
-         input_layout.format == format::bs_fs_zyx_bsv16_fsv16) &&
-        (input_layout.data_type == data_types::f32 || input_layout.data_type == data_types::f16) &&
-        ((input_layout.size.feature[0] / conv->split()) % 16 == 0 || input_layout.size.feature[0] == 3) &&
-        weights_layout.data_type == input_layout.data_type &&
-        (weights_layout.size.batch[0] % 16 == 0 || (weights_layout.size.batch[0] == 8 && conv->groups > 1)) &&
```

Comment:
If I remember correctly, some models with groups and OFM size smaller than 8 execute much slower with `b_fs_zyx_fsv16` layout rather than `bfzyx` format with reorders on reference kernel. Did you check this cases?

Reply:
Yeah, I shouldn't have deleted that condition. I added it back so it's opting this kernel for groups > 1 on the same conditions as before.